.gitlab/download-serverless-artifacts.sh

```diff
@@ -0,0 +1,86 @@
+# This artifacts include: Tracer and ClrProfiler
+#
+
+set -eo pipefail
+
+#Create a directory to store the files
+target_dir=artifacts
+mkdir -p $target_dir
+
+# if [ -n "$CI_COMMIT_TAG" ] || [ -n "$DOTNET_PACKAGE_VERSION" ]; then
+#   echo "Downloading artifacts from Github"
+#   VERSION=${DOTNET_PACKAGE_VERSION:-${CI_COMMIT_TAG##v}} # Use DOTNET_PACKAGE_VERSION if it exists, otherwise use CI_COMMIT_TAG without the v
+
+#   for SUFFIX in "" ".arm64"; do
+#     curl --location --fail \
+#       --output $target_dir/datadog-dotnet-apm-${VERSION}${SUFFIX}.tar.gz \
+#       "https://github.com/DataDog/dd-trace-dotnet/releases/download/v${VERSION}/datadog-dotnet-apm-${VERSION}${SUFFIX}.tar.gz"
+#   done
+
+#   echo -n $VERSION > $target_dir/version.txt
+#   exit 0
+# fi
+
+branchName="refs/heads/$CI_COMMIT_BRANCH"
+
+echo "Looking for azure devops PR builds for branch '$branchName' for commit '$CI_COMMIT_SHA' to start"
+
+# We should _definitely_ have the build by now, so if not, there probably won't be one
+# Check for PR builds first (as more likely to be "full" builds)
+allBuildsForPrUrl="https://dev.azure.com/datadoghq/dd-trace-dotnet/_apis/build/builds?api-version=7.1&definitions=54&\$top=100&queryOrder=queueTimeDescending&reasonFilter=pullRequest"
+buildId=$(curl -sS $allBuildsForPrUrl | jq --arg version $CI_COMMIT_SHA --arg branch $CI_COMMIT_BRANCH '.value[] | select(.triggerInfo["pr.sourceBranch"] == $branch and .triggerInfo["pr.sourceSha"] == $version)  | .id' | head -n 1)
+
+if [ -z "${buildId}" ]; then
+  echo "No PR builds found for commit '$CI_COMMIT_SHA' on branch '$branchName'. Checking for standalone builds..."  
+  allBuildsForBranchUrl="https://dev.azure.com/datadoghq/dd-trace-dotnet/_apis/build/builds?api-version=7.1&definitions=54&\$top=10&queryOrder=queueTimeDescending&branchName=$branchName&reasonFilter=manual,individualCI"
+  buildId=$(curl -sS $allBuildsForBranchUrl | jq --arg version $CI_COMMIT_SHA '.value[] | select(.sourceVersion == $version and .reason != "schedule")  | .id' | head -n 1)
+fi
+
+if [ -z "${buildId}" ]; then
+  echo "No build found for commit '$CI_COMMIT_SHA' on branch '$branchName' (including PRs)"
+  exit 1
+fi
+
+echo "Found build with id '$buildId' for commit '$CI_COMMIT_SHA' on branch '$branchName'"
+
+architectures=("x64" "arm64")
+for architecture in "${architectures[@]}"; do
+  echo "Looking for artifacts for architecture '$architecture'"
+
+  artifacts=("linux-tracer-home-linux-$architecture-r2r" "linux-universal-home-linux-$architecture")
+
+  # Now try to download the artifacts from the build
+  for artifactName in "${artifacts[@]}"; do
+    artifactsUrl="https://dev.azure.com/datadoghq/dd-trace-dotnet/_apis/build/builds/$buildId/artifacts?api-version=7.1&artifactName=$artifactName"
+
+    # Keep trying to get the artifact for 30 minutes
+    downloadUrl=""
+    TIMEOUT=1800
+    STARTED=0
+    until (( STARTED == TIMEOUT )) || [ ! -z "${downloadUrl}" ] ; do
```

Comment:
FYI this will 30mins for the first artifact and then 30 mins for the next one too. That's prob not what we _really_ want, but it's not a big deal, and prob not worth fixing frankly! :D

Reply:
I think this is fine, I tested multiple times, and some artifacts were not available, so I think this is OK